validator/src/main.rs

```diff
@@ -2680,36 +2725,54 @@ pub fn main() {
                     exit(1)
                 })
             });
-    validator_config.snapshot_config = Some(SnapshotConfig {
-        full_snapshot_archive_interval_slots: if snapshot_interval_slots > 0 {
-            snapshot_interval_slots
+
+    let snapshot_interval_slots =
+        value_t_or_exit!(matches, "incremental_snapshot_interval_slots", u64);
+    let is_incremental_snapshot_enabled = matches.is_present("incremental_snapshots");
+    let (full_snapshot_archive_interval_slots, incremental_snapshot_archive_interval_slots) =
+        if snapshot_interval_slots > 0 {
+            if is_incremental_snapshot_enabled {
+                (
+                    value_t_or_exit!(matches, "full_snapshot_interval_slots", u64),
+                    value_t_or_exit!(matches, "incremental_snapshot_interval_slots", u64),
+                )
+            } else {
+                (snapshot_interval_slots, Slot::MAX)
+            }
```

Comment:
It took me a little while to convince myself that there wasn't a bug in this logic. This makes things more readable for me at least:
```suggestion
    let (full_snapshot_archive_interval_slots, incremental_snapshot_archive_interval_slots) =
        let incremental_snapshot_interval_slots = value_t_or_exit!(matches, "incremental_snapshot_interval_slots", u64);      
        if incremental_snapshot_interval_slots > 0 {
            if matches.is_present("incremental_snapshots") {
                (
                    value_t_or_exit!(matches, "full_snapshot_interval_slots", u64),
                    incremental_snapshot_interval_slots,
                )
            } else {
                (incremental_snapshot_interval_slots, Slot::MAX)
            }
```

Reply:
Yep, that looks good to me. I had something like this at one point as well, so happy to go with the version that makes sense to more than just me.